### PR TITLE
readline added

### PR DESCRIPTION
**Pull request by erkinalp**
Readline library from Yorkie Neil added. Might be useful for shell.

### CODE REVIEW

File: Library/readline/readline.c [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+//
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "readline.h"
+
+readline_t *
+readline_new(char * buffer) {
+  readline_t * rl = (readline_t*) malloc(sizeof(readline_t));
```
Comment (Zardoz89) [inline Library/readline/readline.c]: Perhaps it's me, but I have the impression that this library takes too assumptions about that malloc/calloc can't fail.

File: Library/readline/readline.c [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  size_t len = strlen(buffer);
+
+  rl->cursor = 0;
+  rl->line = 0;
+  rl->buffer = (char*) malloc(len);
+  memset(rl->buffer, 0, len);
```
Comment (Zardoz89) [inline Library/readline/readline.c]: Why memset when you are going to overwrite with a memcpy of the same size ?

Comment (erkinalp) [inline Library/readline/readline.c]: It is a dependency-break idiom in superscalar pipelined arches, unnecessary in our case. I will strip it in a later version.

File: Library/readline/readline.c (hunk 1 above)
Comment (erkinalp) [inline Library/readline/readline.c]: *NULL is reset vector on Z80 family so cannot harm any code or data that way on Z80.

Comment (EtchedPixels) [inline Library/readline/readline.c]: Not always true, and since you are doing multiple bytes you then scribble over the vectors.

**Comment by EtchedPixels** [comment]
We have no use for this code in anything in tree or is it part of SVID

**Comment by erkinalp** [comment]
Readline is a dependency for many REPLs despite not being part of SVID or SUS.

**Comment by Zardoz89** [comment]
I'm testing it against GCC .  I find that gives a warning about a bad inlined function, plus one function is broken. I'm trying to fix it...

**Pull request closed without merging**